File: Models/GoodRecipt/GrImportL.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WMS_WEBAPI.Models;

namespace WMS_WEBAPI.Models
{
    [Table("RII_GR_ImportL")]
    public class GrImportL : BaseEntity
    {
        [ForeignKey(nameof(Line))]
        public long? LineId { get; set; }
        public virtual GrLine? Line { get; set; }

        [Required, ForeignKey(nameof(Header))]
        public long HeaderId { get; set; }
        public virtual GrHeader Header { get; set; } = null!;

        [Required, StringLength(35)]
        public string StockCode { get; set; } = null!;

        [StringLength(30)]
        public string? Description1 { get; set; }

        [StringLength(50)]
        public string? Description2 { get; set; }

        public virtual ICollection<GrImportSerialLine> SerialLines { get; set; } = new List<GrImportSerialLine>();
    }
}

File: WebApi/Models/GrImportL.ts
import { BaseEntity } from './BaseEntity';
import { GrLine } from './GrLine';
import { GrHeader } from './GrHeader';
import { GrImportSerialLine } from './GrImportSerialLine';

export interface GrImportL extends BaseEntity {
  lineId?: number;
  line?: GrLine;
  
  headerId: number;
  header?: GrHeader;
  
  stockCode: string;
  description1?: string;
  description2?: string;
  
  // Navigation property for serial lines
  serialLines?: GrImportSerialLine[];
}